packages/ui/lib/components/Form/AddressInput.tsx

```diff
@@ -0,0 +1,178 @@
+      value,
+      className,
+      description,
+      label,
+      withIcon = true,
+      isTruncated,
+      web3Service,
+      localForm,
+      name,
+      ...inputProps
+    } = props
+
+    if (!localForm) return null;
+    const { register } = localForm
+  
+    const [addressType, setAddressType] = useState('')
+    const [resolvedAddress, setResolvedAddress] = useState('')
+    const [resolvedName, setResolvedName] = useState('')
+    const [loadingResolvedAddress, setLoading] = useState(false)
+    const [error, setError] = useState<any>('')
+    const [success, setSuccess] = useState('')
+
+    const inputSizeStyle = SIZE_STYLES[size]
+    let inputStateStyles = ''
+
+    if (error) {
+      inputStateStyles = STATE_STYLES.error
+    } else if (success) {
+      inputStateStyles = STATE_STYLES.success
+    }
+
+    const inputClass = twMerge(
+      'block w-full box-border rounded-lg transition-all shadow-sm border border-gray-400 hover:border-gray-500 focus:ring-gray-500 focus:border-gray-500 focus:outline-none flex-1 disabled:bg-gray-100',
+      inputSizeStyle,
+      inputStateStyles,
+      withIcon ? 'pl-10' : undefined
+    )
+
+    const handleResolver = async (address: string) => {
+      if (address.length) {
+        setLoading(true)
+        setError('')
+        setSuccess('')
+        setAddressType('')
+      }
+      const result = await web3Service.resolveName(address)
+      if (result.name !== null && result.type === 'address') {
+        setLoading(false)
+        setAddressType(result.type)
+        setSuccess(`It's a valid eth address`)
+        setResolvedName(result.name)
+        return result.address;
+      } else if (result.address !== null && result.type === 'name') {
+        setLoading(false)
+        setAddressType(result.type)
+        setSuccess(`It's a valid ens name`)
+        setResolvedAddress(result.address)
+        return result.address;
+      } else if (result.address === null || result.name === null  && result.type === 'error'){
+        setLoading(false)
+        setAddressType('error')
+        setError(`It's not a valid ens name or address`)
+        return result.error;
+      }
+    }
+    return (
+      <>
+        <FieldLayout
+          label={label}
+          size={size}
+          success={success}
+          error={error}
+          description={description}
+        >
+          <div className="flex flex-col">
+            <div className="relative">
+              {withIcon && (
+                <span className="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none">
+                  <Icon size={size} icon={CustomizedIcon} />
+                </span>
+              )}
+              <input
+                {...inputProps}
+                id={label}
+                className={inputClass}
+                {...register(name, {
+                  setValueAs: (value: string) => handleResolver(value),
+                })}
+              />
```

Comment:
`ref` is not set here

Reply:
We don't need to define it since it's already handled by the `register` method.

```
<input 
  onChange={onChange}
  onBlur={onBlur}
  name={name}
  ref={ref}
/>
```

> same as above

```
<input 
  {...register('firstName')} 
/>
```